python/pyarrow/tests/test_types.py

```diff
@@ -44,10 +44,15 @@
     pa.struct([pa.field('a', pa.int32()),
                pa.field('b', pa.int8()),
                pa.field('c', pa.string())]),
+    pa.struct([pa.field('a', pa.int32(), nullable=False),
+               pa.field('b', pa.int8(), nullable=False),
+               pa.field('c', pa.string())]),
     pa.union([pa.field('a', pa.binary(10)),
               pa.field('b', pa.string())], mode=pa.lib.UnionMode_DENSE),
     pa.union([pa.field('a', pa.binary(10)),
               pa.field('b', pa.string())], mode=pa.lib.UnionMode_SPARSE),
+    pa.union([pa.field('a', pa.binary(10), nullable=False),
+              pa.field('b', pa.string())], mode=pa.lib.UnionMode_SPARSE),
     # XXX Needs array pickling
     # pa.dictionary(pa.int32(), pa.array(['a', 'b', 'c'])),
```

Comment:
Unrelated, but what happens if you uncomment this one? Array pickling should work now.

Reply:
Trying that